src/hip_memory.cpp

```diff
@@ -1969,14 +1970,27 @@ hipError_t hipMemGetInfo(size_t* free, size_t* total) {
         }
 
         if (free) {
-            // TODO - replace with kernel-level for reporting free memory:
-            size_t deviceMemSize, hostMemSize, userMemSize;
-            hc::am_memtracker_sizeinfo(device->_acc, &deviceMemSize, &hostMemSize, &userMemSize);
-
-            *free = device->_props.totalGlobalMem - deviceMemSize;
-
-            // Deduct the amount of memory from the free memory reported from the system
-            if (HIP_HIDDEN_FREE_MEM) *free -= (size_t)HIP_HIDDEN_FREE_MEM * 1024 * 1024;
+            hsa_agent_t* agent = static_cast<hsa_agent_t*>(device->_acc.get_hsa_agent());
+            if (!agent){
+		return ihipLogStatus(hipErrorInvalidResourceHandle);
+            } else {
+		uint32_t internal_node_id;
+		hsa_status_t err = hsa_agent_get_info(*agent, (hsa_agent_info_t) HSA_AMD_AGENT_INFO_DRIVER_NODE_ID, &internal_node_id);
+		if(err != HSA_STATUS_SUCCESS) return hipErrorInvalidDevice;
+		
+		size_t deviceMemSize;
+		std::string fileName = std::string("/sys/class/kfd/kfd/topology/nodes/") + std::to_string(internal_node_id) + std::string("/mem_banks/0/used_memory");  
+                
+		std::ifstream file;
+		file.open(fileName);
+		if(!file) return ihipLogStatus(hipErrorFileNotFound);
+		
+		file >> deviceMemSize;    
+		*free = device->_props.totalGlobalMem - deviceMemSize;
```

Comment:
I feel it is good to have a check on read data i.e. deviceMemSize. There is some existing function e.g. isdigit() can be used to make sure read value is indeed a number.

Reply:
used strtol() as the usedmemory can be large,  to check the value read is a number. Updated the PR.